### PR TITLE
Feature/release script

### PR DESCRIPTION
**Pull request by hixi**
Reviewed by:
- [x] @das-g 
- [x] @hixi

### CODE REVIEW

File: release.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import os
+import subprocess
+
+from build_and_push_images import IMAGES, docker_build, docker_push
+
+
+def ask_yes_no(question):
+    if input(question).lower().strip()[0] == "y":
+        return True
+    return False
```
Comment (das-g) [inline release.py]: There's an ~~app~~[standard library function](https://docs.python.org/3/distutils/apiref.html?highlight=distutils.util#distutils.util.strtobool) for this.

Comment (hixi) [inline release.py]: cool, didn't know about that one :-)

File: release.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        ["git", "commit", "-m", 'added makemessages output', 'osmaxx/locale'],
+    ])
+    version_file_path = os.path.join(os.path.dirname(__file__), 'osmaxx', '__init__.py')
+    with open(version_file_path, 'r') as version_file:
+        version_file_content = version_file.readlines()
+    with open(version_file_path, 'w') as version_file:
+        for line in version_file_content:
+            if line.startswith('__version__'):
+                line = "__version__ = '{}'".format(release_version)
+            version_file.write(line)
```
Comment (das-g) [inline release.py]: I didn't know until now, but apparently [`readlines` is considered silly](http://stupidpythonideas.blogspot.ch/2013/06/readlines-considered-silly.html?_escaped_fragment_#!). Probably doesn't matter for a file this small, but you might want to [use `fileinput.input(..., inplace=True)`](http://stackoverflow.com/questions/39086/search-and-replace-a-line-in-a-file-in-python/290494#290494).

File: docs/releasing.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Releasing a new version
+
+Note: We are not using PyPI (for now?) for making releases.
+
+## Prerequisites
+
+- project setup complete, tests running
```
Comment (das-g) [inline docs/releasing.md]: A link to the respective instructions probably wouldn't hurt.

File: release.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    continue_or_stop("You want to make a new release {}".format(release_version))
+
+    prepare_for_release(release_version)
+    make_release_specific_changes(release_version)
+
+    continue_or_stop("Should be continued to push the release to the repository?")
+    release_finish(release_version)
+
+    release = subprocess.check_output(["git", "describe", "--dirty"]).strip().decode()
```
Comment (das-g) [inline release.py]: why are we doing this?

Comment (hixi) [inline release.py]: this should actually be removed...

File: docs/releasing.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Releasing a new version
+
+Note: We are not using PyPI (for now?) for making releases.
+
+## Prerequisites
+
+- project setup complete (see [`README.md` > Docker container bootstrapping > Development](/README.md#development-2))
+- tests running (see [`testing.md` > Setup](testing.md#setup))
```
Comment (das-g) [inline docs/releasing.md]: _passing_ tests migth be a good idea, too

File: release.py [hunk 6]
```diff
@@ -52,13 +52,12 @@ def make_release_specific_changes(release_version):
...
     ])
     version_file_path = os.path.join(os.path.dirname(__file__), 'osmaxx', '__init__.py')
-    with open(version_file_path, 'r') as version_file:
-        version_file_content = version_file.readlines()
-    with open(version_file_path, 'w') as version_file:
-        for line in version_file_content:
-            if line.startswith('__version__'):
-                line = "__version__ = '{}'".format(release_version)
-            version_file.write(line)
+    for line in fileinput.input(version_file_path, inplace=True):
```
Comment (das-g) [inline release.py]: should we use fileinput.input [as a context manager](https://docs.python.org/3/library/fileinput.html#fileinput.input)?

File: .travis.yml [hunk 7]
```diff
@@ -28,8 +28,8 @@ before_install:
 
 install:
   - sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 1397BC53640DB551
-  - sudo apt-get update -qq
-  - sudo apt-get install -qq libspatialite-dev sqlite3 libsqlite3-dev binutils libproj-dev gdal-bin
```
Comment (hixi) [inline .travis.yml]: at least -y should have stayed there ;-)

**Pull request merged**